engines/glk/scott/decompress_z80.h

```diff
@@ -0,0 +1,37 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef GLK_SCOTT_DECOMPRESSZ80
+#define GLK_SCOTT_DECOMPRESSZ80
+
+#include <stdint.h>
```

Comment:
There should be no inclusion of any standard library includes in ScummVM code. Anything the files need from them can probably be obtained from including common\scummsys.h, common\algorithm.h, or others.

Reply:
Fixed. Using `common\scummsys.h` would have meant replacing every instance of `uint8_t` with `uint8` and so on for all types so I instead copied what the AGS engine does and created a new file `types.h` where I include `common\scummsys.h` and typedef over it.

Reply:
One other point I just realized, for consistency, is that the commit prefixes should be "GLK: SCOTT: ", not "GLK SCOTT: ". Would you be able to do an interactive rebase?
If you do: git rebase -i HEAD~22
you should see a list of all the previous commits you did. If you then edit the "pick" before edit commit to be "r" or "reword", then when you save and quit it'll give you a chance to update each commit message in turn to add in the extra colon. Then do a push -f   only when you're happy with the result.

Reply:
Done! 👍